pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -2126,6 +2126,20 @@ double Object_getDoubleValue(JSOBJ Py_UNUSED(obj), JSONTypeContext *tc) {
     return GET_TC(tc)->doubleValue;
 }
 
+const char *Object_getBigNumStringValue(JSOBJ obj, JSONTypeContext *tc, 
+                                    size_t *_outLen) {
+    PyObject* repr = PyObject_Str(obj);
+    PyObject* str = PyUnicode_AsEncodedString(repr, "utf-8", "~E~");
+    char *bytes = PyBytes_AS_STRING(str);
+
+    Py_XDECREF(repr);
```

Comment:
If this object gets released would also free bytes, so need to memcpy first. 

Also just use Py_DECREF for these as simple enough to see lifecycle

Reply:
Ok! Now I memcpy bytes into `tc->cStr` before freeing up `repr`, `str` and `bytes`.

Looking at `tc->cStr` I think it  gets freed-up in `Object_endTypeContext`:
https://github.com/pandas-dev/pandas/blob/c45e92c3956fd2638980ac46e6e93ec3b6cc7c52/pandas/_libs/src/ujson/python/objToJSON.c#L2109